From 3a400115c81ea63530cc558aceffada5bcaf9990 Mon Sep 17 00:00:00 2001
From: "Travis A. Everett" <travis.a.everett@gmail.com>
Date: Tue, 24 Jan 2023 19:26:03 -0600
Subject: disable internal py-yajl for nix-built

---
 build/py.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/py.sh b/build/py.sh
index 6d89eb604..2539bbc3e 100755
--- a/build/py.sh
+++ b/build/py.sh
@@ -411,7 +411,7 @@ py-extensions() {
   fanos
 
   # Require submodule
-  yajl
+  # yajl
 }
 
 minimal() {
-- 
2.39.0

