From a827287bec35c3becbe04369528f903409a99897 Mon Sep 17 00:00:00 2001
From: "Travis A. Everett" <travis.a.everett@gmail.com>
Date: Sat, 28 Jan 2023 18:52:28 -0600
Subject: disable fanos (unused feature)

---
 build/py.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

@@ -360,7 +360,7 @@ py-extensions() {
   pylibc
   line-input
   posix_
-  fanos
+  # fanos
   fastfunc
 }

diff --git a/core/main_loop.py b/core/main_loop.py
index 074a51334..474b6ca6e 100644
--- a/core/main_loop.py
+++ b/core/main_loop.py
@@ -22,7 +22,7 @@ from osh import cmd_eval
 from mycpp import mylib
 from mycpp.mylib import log, print_stderr, tagswitch
 
-import fanos
+# import fanos
 import posix_ as posix
 
 from typing import cast, Any, List, TYPE_CHECKING
-- 
2.36.0

